ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel/Genesis.hs

```diff
@@ -0,0 +1,263 @@
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | An implementation of the prefix selection algorithm.
```

Comment:
I wonder if we would like to write a more comprehensive description of the algorithm we implement, why, how does it fit with the rest of the code. We might want to link to the genesis paper as well.